src/components/AppHighlightAlert/AppHighlightAlertWrapper.jsx

```diff
@@ -2,35 +2,41 @@ import React, { useEffect, useState } from 'react'
 
 import { getAvailableAppHighlightAlerts } from 'components/AppHighlightAlert/helpers'
 import { useClient } from 'cozy-client'
+import log from 'cozy-logger'
 
 const AppHighlightAlertWrapper = ({ apps }) => {
   const [appHighlightAlerts, setAppHighlightAlerts] = useState([])
+  const [isAppHighlightAlertsError, setIsAppHighlightAlertsError] =
+    useState(false)
   const client = useClient()
 
   useEffect(() => {
     const getAppHighlightAlerts = async () => {
-      const availableAppHighlightAlerts = await getAvailableAppHighlightAlerts(
-        client,
-        apps
-      )
-
-      setAppHighlightAlerts(availableAppHighlightAlerts)
+      try {
+        const availableAppHighlightAlerts =
+          await getAvailableAppHighlightAlerts(client, apps)
+
+        setAppHighlightAlerts(availableAppHighlightAlerts)
+      } catch (error) {
+        log('error', `App highlight error: ${error}`)
+        setIsAppHighlightAlertsError(true)
+      }
     }
 
-    if (apps && appHighlightAlerts.length === 0) {
+    if (apps && !isAppHighlightAlertsError && appHighlightAlerts.length === 0) {
       getAppHighlightAlerts()
     }
-  }, [client, apps, appHighlightAlerts.length])
+  }, [client, apps, appHighlightAlerts, isAppHighlightAlertsError])
 
-  useEffect(() => {
+  if (appHighlightAlerts && appHighlightAlerts?.length > 0) {
     appHighlightAlerts.forEach(component => {
-      if (component.displayed) {
-        component.onDisplayed()
-      } else {
-        component.onNotDisplayed()
+      if (component) {
+        component.displayed
+          ? component.onDisplayed()
+          : component.onNotDisplayed()
       }
     })
-  }, [appHighlightAlerts])
+  }
```

Comment:
Calling methods in React components and outside of a hook is considered a bad practice (those methods would be called on every rerender).

Seems like the fix here is on the added `if`. Would it work if keep this `if` but don't remove the `useEffect()`?

Reply:
Fair enough, fixed